### PR TITLE
update top-level docs for Code

### PR DESCRIPTION
**Pull request by khorne3**
This PR:

- adds an overview page for Code
- updates the getting-started page (new title: Scan your codebase)
- updates links to prevent broken ones

### Please ensure

- [x] A technical writer reviews the content or PR
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

**Comment by khorne3** [comment]
@s-santillan what do you think of renaming `docs/semgrep-code/getting-started.md` to `docs/semgrep-code/scan-your-code.md` or similar? I don't think it's absolutely essential, but it would make the URL more informative

File: docs/semgrep-code/overview.md
Comment (s-santillan) [inline docs/semgrep-code/overview.md:6]:
```suggestion
description: Learn about Semgrep Code, a static application security testing (SAST) tool |
```

🙈

File: docs/semgrep-code/overview.md
Comment (s-santillan) [inline docs/semgrep-code/overview.md:7]:
```suggestion
    designed to find security vulnerabilities in your first-party code.
```
 
We can differentiate products through their scan targets:

Semgrep Code - first-party code
Semgrep Supply Chain - third-party code

"applications" could imply both targets but Code specifically scans only 1st party, not the whole "application"

File: docs/semgrep-code/overview.md
Comment (s-santillan) [inline docs/semgrep-code/overview.md:35]: I leave it to you if you want to keep the parenthesis or not, just noting it here.

File: docs/semgrep-code/overview.md
Comment (s-santillan) [inline docs/semgrep-code/overview.md:22]:
```suggestion
In addition to rules available in the [Registry](https://semgrep.dev/r), you can write custom rules to determine what Semgrep Code detects in your repositories. Whether you use pre-existing rules or write custom rules, knowing *which* rules Semgrep Code runs can help you understand how it detects security issues.
```

- The rules in the Registry are not actually "default" - a lot of them are community contributed, some of them are from our SR team, so the quality and freshness can vary. Then there is also the "default" ruleset, which is something else entirely, so I just said more or less that the Registry is where they can find rules.
- There's no particular time that the scan occurred (we're not in a procedure) so I kept it to present tense.

File: docs/semgrep-code/overview.md
Comment (s-santillan) [inline docs/semgrep-code/overview.md:15]:
```suggestion
Semgrep Code is a static application security testing (SAST) tool that inspects your application for security vulnerabilities.

```

File: docs/semgrep-code/overview.md
Comment (s-santillan) [inline docs/semgrep-code/overview.md:9]:
```suggestion
  - Semgrep Code
```

File: docs/semgrep-code/overview.md
Comment (s-santillan) [inline docs/semgrep-code/overview.md:37]:
```suggestion
For cross-file and cross-function analysis (interfile and interprocedural analysis, respectively), you can enable [Semgrep's Pro Engine](/semgrep-code/semgrep-pro-engine-intro/). These scans produce fewer false positives and more true positives, but take longer to complete.
```

File: docs/semgrep-code/overview.md
Comment (khorne3) [reply docs/semgrep-code/overview.md:42]: I think it should be okay (especially if we rename semgrep-code/getting-started to semgrep-code/scan-your-codebase) -- this isn't the quickstart, just in case that's what you're thinking? I do realize there's some redundancy so I plan to refactor and create a Semgrep install instructions component

image

File: docs/semgrep-code/overview.md
Comment (s-santillan) [inline docs/semgrep-code/overview.md:7]:
```suggestion
    that finds security vulnerabilities in your first-party code.
```

Extremely minor nit, you can keep the original wording if you like, this is just a cut-down-words suggestion.

File: docs/semgrep-code/overview.md
Comment (s-santillan) [inline docs/semgrep-code/overview.md:15]:
```suggestion
Semgrep Code is a static application security testing (SAST) tool that detects security vulnerabilities in your first-party code.
```

Just avoiding the use of "application".

File: docs/semgrep-code/overview.md
Comment (s-santillan) [inline docs/semgrep-code/overview.md:31]:
```suggestion
* Customize how Semgrep Code scans your repositories
```

File: docs/semgrep-code/overview.md
Comment (s-santillan) [inline docs/semgrep-code/overview.md:32]:
```suggestion
* Manage your users and facilitate team collaboration in remediating security issues
```
For consistency.

File: docs/semgrep-code/getting-started.md
Comment (s-santillan) [inline docs/semgrep-code/getting-started.md:2]: Hey Katie. FYI, I like that your changes to this doc put the focus on Semgrep **Code**. We probably will repeat this convention of Overview + Scan your codebase for all the products in Semgrep.

Just curious - do we assume that the user has gone through the quickstart? If so, should we not have a "If you have not done so already, set up your Semgrep account by following the steps in Quickstart" or something like that?

File: docs/semgrep-code/getting-started.md
Comment (s-santillan) [inline docs/semgrep-code/getting-started.md:38]: Wonder if it's worth it to move this to the Semgrep Code > Overview doc, which is our "how it works" / [explanation](https://diataxis.fr/) document.

File: src/components/procedure/_platform-add-repo.md
Comment (s-santillan) [inline src/components/procedure/_platform-add-repo.md:12]: I wonder if we should be putting these account + repository requirements in the quickstart, IF we are assuming that the user has gone through the quickstart BEFORE looking at Semgrep Code (and it would save us having redundancy in our docs, I think).

File: docs/semgrep-code/overview.md
Comment (s-santillan) [inline docs/semgrep-code/overview.md:34]: https://developers.google.com/style/word-list#vs
```suggestion
## OSS versus Pro Engine
```

File: docs/semgrep-code/getting-started.md
Comment (s-santillan) [inline docs/semgrep-code/getting-started.md:35]: This is a pretty tricky section. Right now, the steps as-is are not complete. If the user is coming from the quickstart, can you confirm that the new onboarding **includes adding an SCM connection?** If not, we'll need an onboarding document to cover that, so that we can keep **this** doc focused on running the code product specifically (something like `semgrep ci --code` and not just the generic `semgrep ci` command, which will run a code and SCA scan by default.

File: docs/semgrep-code/getting-started.md
Comment (s-santillan) [inline docs/semgrep-code/getting-started.md:39]: I think we can remove this admonition and let the user explore on their own. The demo feature is self-explanatory.

File: docs/semgrep-code/getting-started.md
Comment (s-santillan) [inline docs/semgrep-code/getting-started.md:74]: This section is pretty useful for certain use cases, especially as a fallback for any misconfigurations in `semgrep ci`, but I think we'll have to move it out of this doc - it's not a golden path we want to user to have to wade through. We probably need to add to this PR a dedicated guide to running `semgrep ci` in a local CLI, not a remote environment. I realize it's a very big PR at this point, that's why we're refactoring it.

I can commit directly to this PR to address this, if you're ok with it, without touching any of the suggestions I'm making. LMK if that works for you.

File: docs/semgrep-code/getting-started.md
Comment (s-santillan) [inline docs/semgrep-code/getting-started.md:97]: We probably want to include this in a future onboarding doc. I'm noting this down so that we can collect material already written for an onboarding doc - I'll resolve this comment independently when I've started prepping the PR for it.

File: docs/semgrep-code/getting-started.md
Comment (khorne3) [reply docs/semgrep-code/getting-started.md:2]: That's a good idea; I'll create a component for this, since I think we'll need it for all the top-level product pages

File: docs/semgrep-code/getting-started.md
Comment (khorne3) [reply docs/semgrep-code/getting-started.md:74]: > I can commit directly to this PR to address this, if you're ok with it, without touching any of the suggestions I'm making. LMK if that works for you.

Yes, that's perfectly fine with me!

File: src/components/procedure/_platform-add-repo.md
Comment (khorne3) [reply src/components/procedure/_platform-add-repo.md:12]: I think it's a good idea to move this out, and maybe now's a good time to start a Prerequisites page that we can link to as necessary. It would also link to the supported languages doc, and it would minimize the amount of extra material on the quickstart/product TOFU docs. WDYT?

Comment (khorne3) [reply src/components/procedure/_platform-add-repo.md:12]: IDK, is this too much? Screenshot 2023-12-11 at 2 19 09 PM

Comment (s-santillan) [reply src/components/procedure/_platform-add-repo.md:12]: What would be in the prereq page though? If it's just that callout it'd be pretty short:

>:::info Prerequisites
> * **For GitHub or GitLab SaaS users:** A GitHub or GitLab SaaS repository associated with your account.
> * **For BitBucket SaaS users:** A BitBucket repository and sufficient permissions to edit a BitBucket Pipeline and add repository variables.

Comment (khorne3) [reply src/components/procedure/_platform-add-repo.md:12]: So I think the existing call out is a bit strange. I propose we have a prereqs page that includes this section

```

## CI/CD Systems

Semgrep supports integration with the following CI/CD systems to provide automated scanning of your repositories:

* Azure Pipelines
* Bitbucket
* Buildkite
* CircleCI
* GitHub Actions
* GitLab CI/CD
* Jenkins

When integrating Semgrep into your CI/CD system, ensure that your user account has sufficient permissions to modify pipelines and create and modify environment variables.
```

On the existing page, I have linked to the Add a SCM doc on the SCP page. As it stands, I think the callout is a bit vague and a bit too specific for the general nature of the section (it's the add a repo section, yes, but we talk about non-SCMs here too, like Jenkins)

File: src/components/concept/_semgrep-scan.mdx
Comment (s-santillan) [inline src/components/concept/_semgrep-scan.mdx:7]:
```suggestion
 Semgrep Code uses **rules**, which encapsulate pattern matching logic and data flow analysis, to scan your code and find issues such as code violations, security issues, and the use of outdated or vulnerable libraries. Semgrep generates and reports **findings** to you whenever it finds code that matches the patterns defined by rules.
```

Simplified the English to make it easier for folks to follow. The old text says "...to you findings" - basically I clarified the indirect/direct object.

File: docs/semgrep-code/overview.md
Comment (s-santillan) [reply docs/semgrep-code/overview.md:15]: I think only the first part of this comment (double newline removal) was committed?

File: src/components/procedure/_platform-add-repo.md
Comment (khorne3) [reply src/components/procedure/_platform-add-repo.md:4]: Resolving this, since we're working on prereqs as a separate task

File: docs/semgrep-code/getting-started.md
Comment (s-santillan) [reply docs/semgrep-code/getting-started.md:71]: Update: Sara to make follow up PR instead so we don't add too much to this one and just approve it.

**Comment by s-santillan** [comment]
> @s-santillan what do you think of renaming `docs/semgrep-code/getting-started.md` to `docs/semgrep-code/scan-your-code.md` or similar? I don't think it's absolutely essential, but it would make the URL more informative

Let's update the URL in some other PR (maybe along with the sidebars) so that nothing breaks for now :)

**Review by s-santillan — APPROVED** [review]
Approving with plans for follow up PRs!!

**Pull request merged**
